src/eval/operation.rs

```diff
@@ -2559,18 +2568,40 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                     with {
                         Term::Record(record_data) => {
                             let contract = contract_term.closurize(&mut self.cache, &mut env2, contract_env);
-                            let record_contract = Term::Record(RecordData{
-                                fields: record_data.fields.iter().map(|(key, value)| (*key, Field {
-                                    value: None,
-                                    pending_contracts: vec![PendingContract::new(contract.clone(), lbl.clone())],
-                                    metadata: FieldMetadata {
-                                        opt: value.metadata.opt,
-                                        ..Default::default()
-                                    },
-                                })).collect(),
-                                attrs: RecordAttrs{
-                                    open: true
-                                },
+                            let record_contract = Term::Record(RecordData {
+                                fields: record_data
+                                    .fields
+                                    .iter()
+                                    .map(|(key, value)| {
+                                        (
+                                            *key,
+                                            Field {
+                                                value: None,
+                                                pending_contracts: vec![PendingContract::new(
+                                                    contract.clone(),
+                                                    lbl.clone(),
+                                                )],
+                                                // This is intentionally not written as
+                                                // `{ opt: value.metadata.opt, ..Default::default() }`
+                                                //
+                                                // When `FieldMetadata` is changed
+                                                // in the future, we will need to consider
+                                                // which values should be inherited from
+                                                // `value.metadata` and which values can be
+                                                // left at their defaults.
```

Comment:
I think it's still missing the important information: `metadata` must be choosen such that, when merged with the original fields' metadata, all but `annotation.contracts` must be left unchanged.

Reply:
In fact, the metadata should be left entirely unchanged upon merging. The contract is directly inserted into the `pending_contracts` field because the primop is getting a closure, not just a term.

Reply:
Hmm. I wonder if we should still insert it, so that `nickel query` shows all contracts attached (otherwise could be strange that you get a contract failure but can't see the contract in the list given by `query`). We can use the type contained in the label, which should be untransformed. That being said, let's think about that separately, as this PR is already approved and reviewed.